src/audio/effects.c

```diff
@@ -189,23 +189,33 @@ s32 get_vibrato_pitch_change(struct VibratoState *vib) {
 #endif
 
 f32 get_vibrato_freq_scale(struct VibratoState *vib) {
+    s32 vibratoExtentTarget;
+
     if (vib->delay != 0) {
         vib->delay--;
         return 1;
     }
 
+    // This needs to be set locally because changing the original value to 0 overrides the whole channel,
+    // effectively negating the ability to disable vibrato and bypass this function in the first place.
+    // This function isn't huge but it otherwise would get called many thousands of times per second.
+    vibratoExtentTarget = vib->seqChannel->vibratoExtentTarget;
```

Comment:
```suggestion
    if (vib->delay != 0) {
        vib->delay--;
        return 1;
    }

    // This needs to be set locally because changing the original value to 0 overrides the whole channel,
    // effectively negating the ability to disable vibrato and bypass this function in the first place.
    // This function isn't huge but it otherwise would get called many thousands of times per second.
    s32 vibratoExtentTarget = vib->seqChannel->vibratoExtentTarget;
```
could you just set it here?

Reply:
I guess I could, though I actually think I like the look of predefining variables at the top of functions more than mid-function, depending on the situation. I don't _think_ it compiles any differently?

In this case, I like it predefined since it's being used as a temporary variable and I want it to be clear from the start that it shouldn't get confused with use of `vib->seqChannel->vibratoExtentTarget`, though I wouldn't say I live or die by it in this example.